x-pack/plugins/alerting/server/task_runner/execution_handler.ts

```diff
@@ -198,6 +198,15 @@ export class ExecutionHandler<
           continue;
         }
 
+        if (alert && alert.getMaintenanceWindowIds().length > 0) {
+          this.logger.debug(
+            `no scheduling of actions "${action.id}" for rule "${
+              this.taskInstance.params.alertId
+            }": has active maintenance windows ${alert.getMaintenanceWindowIds()}.`
+          );
+          continue;
+        }
```

Comment:
Can you move this check into `generateExecutables`  method please?
We create executables `(action + alert or summarizedAlerts)` array in there and iterate through them and schedule here. We skip them here only for rate limiting, so we can log how many actions were created and how may of them have been skipped.

Reply:
Moving this to just after `for (const [alertId, alert] of alertsArray) {` should be enough.

Reply:
> Can you move this check into generateExecutables method please?

We decided to have the code here because we still want to count up these actions as `generated actions` via  `ruleRunMetricsStore.incrementNumberOfGeneratedActionsByConnectorType(actionTypeId);`. We can move it 
if we don't care abut this count, @XavierM what do you think? 

> AFAIU you want to send notifications for recovered alerts if the alert was created before the maintenance window. In order to have that check, you may need to add if(!this.isRecoveredAlert(action.group)) condition as well.

We make this determination using the `maintenance_window_ids` field, so we don't need to check `isRecoveredAlert`, any alerts marked with a maintenance window id will never send notifications, recovered alerts that originated outside a window, but recovered during a window, it will still send notifications.